### PR TITLE
Jetpack: Add block and allow list toggles to WAF settings

### PR DESCRIPTION
**Pull request by dkmyta**
### Description
Updates the Jetpack WAF security settings UI to allow the IP block and allow lists to be toggled independently.

## Proposed changes:

* Makes use of the new `jetpack_waf_ip_allow_list_enabled` / `jetpack_waf_ip_block_list_enabled` settings, replacing `jetpack_waf_ip_list`
* Removes the individual manual rules setting and adds independent toggles for each IP list
* Modifies sections dependent on the original setting

### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1263

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch
* Start up JT, and enable Jetpack
* Proceed to WAF security settings and toggle the various options in different combinations
* Ensure the settings all perform as expected and that no regressions are introduced
* Upgrade and downgrade and ensure all expected settings are still accessible and functional
* Verify that a blocklisted IP cannot make requests or access the site admin when BFP is enabled
* Ensure that the blocked IP can once again do so once whitelisted
* Ensure cross compatibility exists when Protect is enabled and the settings are toggled from that end
* Ensure backward compatibility exists when the update is applied to an existing site with the prior settings in place

## Screenshots

All Enabled:
Screen Shot 2024-07-17 at 06 47 41

All Disabled:
Screen Shot 2024-07-17 at 06 46 54

WAF Disabled:
Screen Shot 2024-07-17 at 06 46 02

Unsupported:
Screen Shot 2024-07-17 at 06 45 08

### CODE REVIEW

**Comment by nateweller** [comment]
Similar to https://github.com/Automattic/jetpack/pull/38265, what do you think about using this PR to also de-duplicate the allow list?

We could place the allow list as a separate card below the WAF and BFP cards, keeping/revising the description included with the BFP allow list.

File: projects/plugins/jetpack/_inc/client/security/allowList.jsx [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+					? allowListInputState
+					: getSetting( state, 'jetpack_waf_ip_allow_list' ),
+			isFetchingSettings: isFetchingWafSettings( state ),
+			isUpdatingWafSettings: isUpdatingWafSettings( state ),
+			settings: getWafSettings( state ),
+		};
+	},
+	dispatch => {
+		return {
+			updateWafIpAllowList: allowList => dispatch( updateWafIpAllowList( allowList ) ),
```
Comment (nateweller) [inline projects/plugins/jetpack/_inc/client/security/allowList.jsx:263]:
```suggestion
```

Unused

**Review by nateweller — CHANGES_REQUESTED** [review]
I think the `updateWafIpAllowList` and `allowListInputState` logic can be removed, as it is not necessary now that we aren't duplicating the list on the page - we could just use `updateWafSettings` and `getWafSettings`, in addition to tracking the current text input value in local component state :+1:

**Comment by dkmyta** [comment]
>we could just use updateWafSettings and getWafSettings, in addition to tracking the current text input value in local component state

Very good point! Updated accordingly 👍🏻

**Review by nateweller — APPROVED** [review]
LGTM! We should wait on https://github.com/Automattic/jetpack/pull/38395 to merge though :+1:

**Pull request merged**